clusters/resource_cluster.go

```diff
@@ -58,6 +60,59 @@ func ZoneDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
 	return false
 }
 
+func Validate(cluster compute.CreateCluster) error {
```

Comment:
nit: for the duplicated functions, leave a comment here and on the original functions to link them to one another while both exist, so that people know that changes need to be made in both.